File: leetcode/longestConsecutiveSeq/solution1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        sort(nums.begin(), nums.end());
        int n = 0,max = 0;
        for(int i = 0;i < nums.size() - 1;i++) {
            if(nums[i] == nums[i+1]) continue;
            if((nums[i+1] - nums[i]) == 1) n++;
            else n = 0;
            if(n > max) max = n;
        }
        return max + 1;
    }
};

File: leetcode/maxErasureVal/solution1.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        int i = 0;
        int j = 0;
        int str = 0;
        int max = 0;
        int n = nums.length;
        HashMap<Integer, Integer> map = new HashMap<>();

        while(j < n) {
            int chAtJ = nums[j];

            map.put(chAtJ, map.getOrDefault(chAtJ, 0) + 1);
            str += chAtJ;

            while((j - i + 1) != map.size()) {
                int chAtI = nums[i];

                map.put(chAtI, map.getOrDefault(chAtI, 0) - 1);
                str -= chAtI;

                if(map.get(chAtI) <= 0) {
                    map.remove(chAtI);
                }

                i++;
            }

            max = Math.max(max, str);
            j++;
        }

        return max;
    }
}